src/usr/local/www/guiconfig.inc

```diff
@@ -29,8 +29,22 @@ if (!$nocsrf) {
 	function csrf_startup() {
 		global $config;
 		csrf_conf('rewrite-js', '/csrf/csrf-magic.js');
-		$timeout_minutes = isset($config['system']['webgui']['session_timeout']) ? $config['system']['webgui']['session_timeout'] : 240;
-		csrf_conf('expires', $timeout_minutes * 60);
+
+//FIXME: 
+// This code doesn't seem to work as expected, because $config[] isn't accessible within csrf_startup() when it executes, so it mis-sets csrf timeout. 
+// Ref - https://redmine.pfsense.org/issues/6803 (1st note)
+
+		if (!isset($config['system']['webgui']['session_timeout'])) {
+			csrf_conf('expires', 240 * 60);
```

Comment:
It may not be able to access $config at this point, but can it not access $g either? Seems it would be better to read the value straight from $g so it doesn't need to be put in two places, but if it's too early for that then there may not be any better choice.

Thinking more generally, there is also the option of reading it via a shell exec to /usr/local/sbin/read_xml_tag.sh at some point during the boot process and then setting it in an environment variable or in some other readable way. Doing that directly from PHP is ugly and parsing the config in PHP at that point also seems ugly since it will be read in shortly anyhow.